R/utilities-plots.R

```diff
@@ -79,6 +79,79 @@ autoAxesTicksFromLimits <- function(limits) {
   return(rep(c(1, 2, 5), length(logTicks)) * 10^rep(logTicks, each = 3))
 }
 
+#' @title getPlotConfigurationFromPlan
+#' @description Get the appropriate `PlotConfiguration` object with scaled dimensions for exporting it
+#' @param plotProperties Plot properties from configuration plan
+#' @param plotType Name of plot type to call the appropriate `PlotConfiguration` object.
+#' E.g. for pk ratio plots, use "PKRatio" to create a `PKRatioPlotConfiguration` object
+#' @param legendPosition Legend position in order to add scale factor in the final plot dimensions 
+#' that accounts for possible shrinking of the plot panel due to the addition of the legend
+#' @return A `PlotConfiguration` object
+#' @keywords internal
+getPlotConfigurationFromPlan <- function(plotProperties, plotType = NULL, legendPosition = reEnv$theme$background$legendPosition) {
+  # Define the appropriate configuration from plotType
+  # by creating expression: "tlf::<plotType>PlotCOnfiguration$new()"
+  plotConfiguration <- eval(parse(text = paste0("tlf::", plotType, "PlotConfiguration$new()")))
+  
+  # Set properties from FontAndSize
+  fonts <- plotProperties$FontAndSize$Fonts

```

Comment:
@pchelle  In the version of Qualification-DDI-CYP3A4 I have there is no `FontAndSize` field under the plot settings.  Has this changed for all configuration plans?

![image](https://user-images.githubusercontent.com/18409098/144306874-1b74e81a-0b0b-4fc5-af80-822b83ff57be.png)

Reply:
Good catch !
In the global settings there is not a `FontAndSize` field, but there is one in local settings for time profile plots.
I'm going to try to generalize that and account for both

Reply:
Actually, the Plot Settings here are global settings sent to the current theme properties when loading/updating the configuration plan (see [`ConfigurationPlan$updateTheme()`](https://github.com/Open-Systems-Pharmacology/OSPSuite.ReportingEngine/blob/a5492d694ad0a384561c022cf3633d8c364cfb06/R/configuration-plan.R#L251)).

The function `getPlotConfigurationFromPlan()` is however called locally. In such case, I only found plot settings for time profile plots which looked like this:
![image](https://user-images.githubusercontent.com/45343665/144384075-fc1c7339-f912-4651-9451-a6e35f3ef688.png)

Since some settings - e.g. for axes, background or legend - were sometimes defined outside `FontAndSize`, I used the upper level field as input of `getPlotConfigurationFromPlan`. This would allow more flexibility if new fields or field names are used in different plots. 

I did not apply the scaling of font at the theme level, I am going to apply it anyways when re-allocating the font properties.

Reply:
just tried it - yes this solution works great!